## Repeatmasking
When annotating a genome assembly, the first step is usually to mask the repeats. This can be thought of as a two-step process. First find the repeats, and then mask them. 'Repeats' in this context means all different kinds of sequences that are repeated across a genome. This might be interspered repeats such as transposable elements which spread themselves across a genome, or tandem repeats, such as short tandem repeats which consists of a motif repeated in tandem (for instance 'AC' repeated six times: ACACACACACAC). There might be multiple reasons for masking repeats, but the two main ones would be to avoid creating a lot of alignments due to the repetitive sequence itself, and to avoid annotating genes found in transposable elements as the genes of the host. Arguably, it could be interesting to actually annotate the genes in transposable elements properly, so to better characterize them, but this is usually outside the scope of a genome annotation project. 

The actual masking itself can be done either hard or soft. Hard masking would entail replacing the found repeats with either 'X' or 'N', thereby hiding the sequence itself and making it unavailable for alignments. The other is soft masking. Nucleotides in fasta files are usually in capital letters (ACTG) and with soft masking the nucleotides in repeats would be replaces with the same, but in lower case (actg). This makes it possible for different tools to avoid creating alignments in these repeats, but alignments can be extended into the repeats. Some tools, such as [miniprot](https://github.com/lh3/miniprot), ignores soft masking (that is, it has no effect on the program). 
