## Protein evidence
[Miniprot](https://github.com/lh3/miniprot) might be the most exciting that has happened in the genome annotation world for many years. Miniprot ([Li 2023](https://academic.oup.com/bioinformatics/article/39/1/btad014/6989621), which is a nice paper on its own, worth a read) maps proteins to a (nucleotide) genome, and basically creates gene models based on the mapped proteins. Since it is so fast, you can quickly map some proteins to a genome and get an impression of the genic content. For closely related species you might not need anything else than mapping the predicted proteins from one (well-assembly and well-annotated) species to another. However, since it is so fast, we like to map different datasets to the genome and merge the resulting aligments with [EvidenceModeler](04_evidencemodeler). We usally use proteins from a well-annotated and well-assembled relatively closely related species, for instance human for mammals. Further, [UniProtKB/Swiss-Prot](https://academic.oup.com/nar/article/51/D1/D523/6835362) is a database with curated protein models. This can be a good source of evidence. Lastly, we often align the relevant part of [OrthoDB v11](https://academic.oup.com/nar/article/51/D1/D445/6814468). This is a large dataset, and might take substantial time and computational power, so we skip aligning OrthoDB proteins here.


  
